### PR TITLE
[image_picker] Add image_picker_tizen package

### PR DESCRIPTION
**Pull request by bwikbs**

### CODE REVIEW

**Comment by bwikbs** [comment]
- Require 5.5 or higher Tizen API
- It has been verified to work only on ``mobile(TM1)``   (Wearable requires a separate gallery app.)

**Comment by swift-kim** [comment]
Also we haven't yet decided how to handle TV-specific builds (`TV_PROFILE`). I'm planning to set the variable in the flutter-tizen tool when a device ID is specified (`flutter-tizen run`) or an option is enabled (`flutter-tizen build tpk --target-profile tv`). Any thoughts?

**Comment by bwikbs** [comment]
You are right. I haven't organized my thoughts yet. I think we need more discussion regarding profile issue.

File: packages/image_picker/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+Then invoke the static `image_picker` method anywhere in your Dart code.
+
+``` dart
+final pickedFile = await picker.getImage(source: ImageSource.gallery);
+```
+
+## Limitations
+
+- This plugin is only supported on **Galaxy Watch** devices running Tizen 5.5 or later.
```
Comment (swift-kim) [inline packages/image_picker/README.md:25]: How about adding this line:
```markdown
- The only API you can use is `getImage(source: ImageSource.gallery)`. You can't pick a video file (`getVideo()`) or pick a file from `ImageSource.camera`.
```

File: packages/image_picker/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Import the library.
+
+``` dart
+import 'package:image_picker/image_picker.dart';
+```
+
+Then invoke the static `image_picker` method anywhere in your Dart code.
+
+``` dart
+final pickedFile = await picker.getImage(source: ImageSource.gallery);
```
Comment (swift-kim) [inline packages/image_picker/README.md:20]: `picker` isn't defined.

```dart
final picker = ImagePicker();
```

File: packages/image_picker/LICENSE [hunk 3]
```diff
@@ -0,0 +1,26 @@
+Copyright (c) 2021 Samsung Electronics Co., Ltd. All rights reserved.
+Copyright (c) 2017 The Chromium Authors. All rights reserved.
```
Comment (swift-kim) [inline packages/image_picker/LICENSE:2]:
```suggestion
Copyright (c) 2019 The Flutter Authors. All rights reserved.
```

File: packages/image_picker/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+name: image_picker_tizen
+description: Flutter plugin for selecting images from the Tizen image
+  library, and taking new pictures with the camera.
+version: 0.0.1
```
Comment (swift-kim) [inline packages/image_picker/pubspec.yaml:4]:
```suggestion
version: 1.0.0
```
For consistency with CHANGELOG.md.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+#include "image_picker_tizen_plugin.h"
+
+#include <app_control.h>
+#include <assert.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar.h>
+#include <flutter/standard_method_codec.h>
+#include <system_info.h>
+
+#include "image_resize.h"
```
Comment (swift-kim) [inline packages/image_picker/tizen/src/image_picker_tizen_plugin.cc:14]: You may move this line to line 25 and also move `tizen/inc/image_resize.h` to `tizen/src/image_resize.h`.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+      } else {
+        SendResultWithError("Invalid arguments");
+        return;
+      }
+
+      if (source == ImageSource::CAMERA) {
+        // TODO: we need to check this feature after webcam is prepared
+        SendResultWithError("Not supported on this device");
+      } else if (source == ImageSource::GALLERY) {
+#ifdef TV_PROFILE
```
Comment (swift-kim) [inline packages/image_picker/tizen/src/image_picker_tizen_plugin.cc:165]: You may remove this condition and just use `CheckPermissionAndPickImage()`.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+#endif
+      } else {
+        SendResultWithError("Invalid video source");
+      }
+    } else {
+      result_->NotImplemented();
+      result_ = nullptr;
+    }
+  }
+  void CheckPermissionAndPickImage() {
```
Comment (swift-kim) [inline packages/image_picker/tizen/src/image_picker_tizen_plugin.cc:178]: Please add linebreaks between functions (there are a few more).

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+  void SendResultWithError(std::string errorCode,
+                           std::string errorMessage = "") {
+    if (result_ == nullptr) {
+      return;
+    }
+    result_->Error(errorCode, errorMessage);
+    result_ = nullptr;
+  }
+
+  ImageResize imageResize_;
```
Comment (swift-kim) [inline packages/image_picker/tizen/src/image_picker_tizen_plugin.cc:301]: For naming convention?
```suggestion
  ImageResize image_resize_;
```

File: packages/image_picker/tizen/src/image_resize.cc [hunk 9]
```diff
@@ -0,0 +1,271 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Copyright 2019 The Flutter Authors. All rights reserved.
```
Comment (swift-kim) [inline packages/image_picker/tizen/src/image_resize.cc:2]: Where did you get this file originally from?

**Review by swift-kim — COMMENTED** [review]
Thanks!

File: packages/image_picker/tizen/src/image_resize.cc (hunk 9 above)
Comment (seungsoo47) [reply packages/image_picker/tizen/src/image_resize.cc:2]: I referred to ImageResizer.java of Android port.

Comment (swift-kim) [reply packages/image_picker/tizen/src/image_resize.cc:2]: Thanks!

File: packages/image_picker/README.md (hunk 1 above)
Comment (bwikbs) [reply packages/image_picker/README.md]: done.

File: packages/image_picker/README.md (hunk 2 above)
Comment (bwikbs) [reply packages/image_picker/README.md:21]: done.

File: packages/image_picker/LICENSE (hunk 3 above)
Comment (bwikbs) [reply packages/image_picker/LICENSE]: done.

File: packages/image_picker/pubspec.yaml (hunk 4 above)
Comment (bwikbs) [reply packages/image_picker/pubspec.yaml]: done.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc (hunk 5 above)
Comment (bwikbs) [reply packages/image_picker/tizen/src/image_picker_tizen_plugin.cc]: done.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc (hunk 6 above)
Comment (bwikbs) [reply packages/image_picker/tizen/src/image_picker_tizen_plugin.cc]: done.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc (hunk 8 above)
Comment (bwikbs) [reply packages/image_picker/tizen/src/image_picker_tizen_plugin.cc]: done.

File: packages/image_picker/tizen/src/image_picker_tizen_plugin.cc (hunk 7 above)
Comment (bwikbs) [reply packages/image_picker/tizen/src/image_picker_tizen_plugin.cc]: done.

File: packages/image_picker/README.md (hunk 1 above)
Comment (swift-kim) [reply packages/image_picker/README.md]: Please append the line but do not remove the existing one. It's still necessary.

Comment (bwikbs) [reply packages/image_picker/README.md:26]: done.

Comment (swift-kim) [reply packages/image_picker/README.md:26]: Thank you!

**Review by swift-kim — APPROVED** [review]

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**